msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase1561081.java

```diff
@@ -0,0 +1,219 @@
+                .msalConfigResourceId(getConfigFileResourceId())
+                .build();
+
+        //AT interactive acquisition.
+        final MsalAuthResult authResult = msalSdk.acquireTokenInteractive(authTestParams_firstTry, new OnInteractionRequired() {
+            @Override
+            public void handleUserInteraction() {
+                final MicrosoftStsPromptHandlerParameters promptHandlerParameters = MicrosoftStsPromptHandlerParameters.builder()
+                        .prompt(PromptParameter.LOGIN)
+                        .loginHint(username)
+                        .sessionExpected(false)
+                        .consentPageExpected(false)
+                        .build();
+
+                new MicrosoftStsPromptHandler(promptHandlerParameters)
+                        .handlePrompt(username, password);
+            }
+        }, TokenRequestTimeout.MEDIUM);
+
+        authResult.assertSuccess();
+
+        //extract the device id claim from the access token.
+        String deviceId = (String) JWTParserFactory.INSTANCE.getJwtParser().parseJWT(authResult.getAccessToken()).get("deviceid");
+
+        //this gets the deviceId from the Ui and matches it to the deviceID obtained from the AT
+        getDeviceIdFromMyAccount(deviceId, username, password, true);
+
+        //init the release candidate BrokerHost in order to run wpj Leave
+        BrokerHost brokerHost = new BrokerHost(BrokerHost.BROKER_HOST_APK_RC);
+        if(brokerHost.isInstalled()){
+            brokerHost.uninstall();
+        }
+
+        brokerHost.install();
+        //run wpj leave
+        brokerHost.wpjLeave();
+
+        final MsalAuthTestParams authTestParams_SecondTry = MsalAuthTestParams.builder()
+                .activity(mActivity)
+                .loginHint(username)
+                .scopes(Arrays.asList(mScopes))
+                .promptParameter(Prompt.LOGIN)
+                .msalConfigResourceId(getConfigFileResourceId())
+                .build();
+
+
+        //acquire the token interactively for a second time.
+         final MsalAuthResult authResult2 = msalSdk.acquireTokenInteractive(authTestParams_SecondTry, new OnInteractionRequired() {
+            @Override
+            public void handleUserInteraction() {
+                final MicrosoftStsPromptHandlerParameters promptHandlerParameters = MicrosoftStsPromptHandlerParameters.builder()
+                        .prompt(PromptParameter.LOGIN)
+                        .loginHint(username)
+                        .sessionExpected(false)
+                        .consentPageExpected(false)
+                        .build();
+
+                new MicrosoftStsPromptHandler(promptHandlerParameters)
+                        .handlePrompt(username, password);
+            }
+        }, TokenRequestTimeout.MEDIUM);
+
+        authResult2.assertSuccess();
+
+        //wait for two minutes to ensure the changes are made.
+        Thread.sleep(TimeUnit.MINUTES.toMinutes(2));
```

Comment:
What is the rationale for choosing two minutes?

Reply:
To give it more than enough time to have the changes propagated in order to have a consistent response on getting the device Id from account.